.github/workflows/publish.yml

```diff
@@ -18,41 +18,18 @@ jobs:
       id-token: write
 
     steps:
-      - uses: actions/checkout@v3
-        with:
-          submodules: recursive
+      - uses: actions/checkout@v4
 
-      - name: Install Python 3
-        uses: actions/setup-python@v3
-        with:
-          python-version: "3.13"
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+
+      - name: Check lockfile
+        run: uv lock --check
 
       - name: Setup Graphviz
         uses: ts-graphviz/setup-graphviz@v1
 
-      - name: Install dependencies
+      - name: Build and publish to pypi
         run: |
-          python -m pip install --upgrade pip
-          python -m pip install -r requirements.txt
-
-      - name: Install pypa/build
-        run: python -m pip install build --user
-
-      - name: Build a binary wheel and a source tarball
-        run: python -m build --sdist --wheel --outdir dist/ .
-
-      - name: Publish distribution to Test PyPI
-        uses: pypa/gh-action-pypi-publish@release/v1
-        with:
-          repository_url: https://test.pypi.org/legacy/
-          # The pypa/gh-action-pypi-publish action sources attestations from the
-          # same source, so leaving attestations enabled (the default behavior)
-          # for both steps will cause the production PyPI step to fail; however,
-          # disabling attestations on the test PyPI step should allow the
-          # production PyPI step to succeed
-          attestations: false
-
-      - name: Publish distribution to PyPI
-        uses: pypa/gh-action-pypi-publish@release/v1
-        with:
-          repository_url: https://upload.pypi.org/legacy/
+          uv build
+          uv publish
```

Comment:
This should just work according to the docs here: https://docs.astral.sh/uv/guides/package/#publishing-your-package

Reply:
@caleb531 do you think we should merge to develop and then debug on the fly during the next deployment? Then some of the other PRs can get merged in the meantime.

Reply:
@eliotwrobson I think that's fine. I'm reading that `uv publish` is compatible with the Trusted Publisher functionality in PyPI, so I feel optimistic about switching to `uv` for publishing. But we won't really know until we can test a publish from `develop`.

Reply:
That makes sense. I'll go ahead and mark this as ready for review.